src/ln/channelmanager.rs

```diff
@@ -4397,6 +4397,69 @@ mod tests {
 		check_added_monitors!(nodes[1], 1);
 	}
 
+	#[test]
+	fn test_update_fee_that_funder_cannot_afford() {
+		let nodes = create_network(2);
+		let chan = create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 10000, 8000);
+		let channel_id = chan.2;
+
+		macro_rules! get_feerate {
+			($node: expr) => {{
+				let chan_lock = $node.node.channel_state.lock().unwrap();
+				let chan = chan_lock.by_id.get(&channel_id).unwrap();
+				chan.get_feerate()
+			}}
+		}
+
+		let feerate = get_feerate!(nodes[0]);
+		nodes[0].node.update_fee(channel_id, feerate+20).unwrap();
+		check_added_monitors!(nodes[0], 1);
+
+		let events_0 = nodes[0].node.get_and_clear_pending_msg_events();
+		assert_eq!(events_0.len(), 1);
+		let (update_msg, commitment_signed) = match events_0[0] {
+			MessageSendEvent::UpdateHTLCs { node_id:_, updates: msgs::CommitmentUpdate { update_add_htlcs:_, update_fulfill_htlcs:_, update_fail_htlcs:_, update_fail_malformed_htlcs:_, ref update_fee, ref commitment_signed } } => {
+				(update_fee.as_ref(), commitment_signed)
+			},
+			_ => panic!("Unexpected event"),
+		};
+		nodes[1].node.handle_update_fee(&nodes[0].node.get_our_node_id(), update_msg.unwrap()).unwrap();
+
+		nodes[1].node.handle_commitment_signed(&nodes[0].node.get_our_node_id(), commitment_signed).unwrap();
+		let (revoke_msg, commitment_signed) = get_revoke_commit_msgs!(nodes[1], nodes[0].node.get_our_node_id());
+		check_added_monitors!(nodes[1], 1);
+
+		nodes[0].node.handle_revoke_and_ack(&nodes[1].node.get_our_node_id(), &revoke_msg).unwrap();
+		assert!(nodes[0].node.get_and_clear_pending_msg_events().is_empty());
+		check_added_monitors!(nodes[0], 1);
+
+		nodes[0].node.handle_commitment_signed(&nodes[1].node.get_our_node_id(), &commitment_signed).unwrap();
+		let revoke_msg = get_event_msg!(nodes[0], MessageSendEvent::SendRevokeAndACK, nodes[1].node.get_our_node_id());
+		// No commitment_signed so get_event_msg's assert(len == 1) passes
+		check_added_monitors!(nodes[0], 1);
+
+		nodes[1].node.handle_revoke_and_ack(&nodes[0].node.get_our_node_id(), &revoke_msg).unwrap();
+		assert!(nodes[1].node.get_and_clear_pending_msg_events().is_empty());
+		check_added_monitors!(nodes[1], 1);
+
+		nodes[0].node.update_fee(channel_id, feerate+2000).unwrap();
```

Comment:
It would be nice to check the boundary condition here - can you make the first fee update one satoshi shy of hitting the feerate limit (and check that it is by getting the latest local commitment transaction out of the channel via Channel::last_local_commitment_txn) and then check that one more breaks it?

Reply:
Done. See the initial channel funds and fees so it is exactly on the limit and then increase by 1 to break it.